### ISSUE TITLE
Craft 4 migration failed m220308_100000_owners_table

### ISSUE THREAD

**Issue by MarcelWeidum** [engram-design/SuperTable#488]
### Describe the bug

When I update the plugin from `2.7.2` to `^3.0` (3.0.2) and run the migrations, the migration `m220308_100000_owners_table` is failing.

Full error:
```
Backing up the database ... done
*** applying m220308_100000_owners_table
    > dropping {{%supertableblocks_owners}} if it exists ... done (time: 0.006s)
    > create table {{%supertableblocks_owners}} ... done (time: 0.009s)
    > add foreign key fk_fqcqwaszmrsorcjrgahqfowopgjdbmwgyvoa: {{%supertableblocks_owners}} (blockId) references {{%supertableblocks}} (id) ... done (time: 0.017s)
    > add foreign key fk_uhjohoezdbxejmwdwwzxsjgdrhcpcyrcqfrc: {{%supertableblocks_owners}} (ownerId) references {{%elements}} (id) ... done (time: 0.017s)
    > update in {{%supertableblocks}} ... done (time: 0.020s)
    > execute SQL: INSERT INTO {{%supertableblocks_owners}} ([[blockId]], [[ownerId]], [[sortOrder]]) 
SELECT [[id]], [[ownerId]], [[sortOrder]] 
FROM {{%supertableblocks}} ...Exception: SQLSTATE[23000]: Integrity constraint violation: 1452 Cannot add or update a child row: a foreign key constraint fails (`database`.`supertableblocks_owners`, CONSTRAINT `fk_uhjohoezdbxejmwdwwzxsjgdrhcpcyrcqfrc` FOREIGN KEY (`ownerId`) REFERENCES `elements` (`id`) ON DELETE CASCADE)
The SQL being executed was: INSERT INTO `supertableblocks_owners` (`blockId`, `ownerId`, `sortOrder`) 
SELECT `id`, `ownerId`, `sortOrder` 
FROM `supertableblocks` (/home/user/domain.com/public_html/vendor/yiisoft/yii2/db/Schema.php:676)
#0 /home/user/domain.com/public_html/vendor/yiisoft/yii2/db/Command.php(1307): yii\db\Schema->convertException()
#1 /home/user/domain.com/public_html/vendor/yiisoft/yii2/db/Command.php(1102): yii\db\Command->internalExecute()
#2 /home/user/domain.com/public_html/vendor/yiisoft/yii2/db/Migration.php(219): yii\db\Command->execute()
#3 /home/user/domain.com/public_html/vendor/verbb/super-table/src/migrations/m220308_100000_owners_table.php(38): yii\db\Migration->execute()
#4 /home/user/domain.com/public_html/vendor/craftcms/cms/src/db/Migration.php(49): verbb\supertable\migrations\m220308_100000_owners_table->safeUp()
#5 /home/user/domain.com/public_html/vendor/yiisoft/yii2/console/controllers/BaseMigrateController.php(758): craft\db\Migration->up()
#6 /home/user/domain.com/public_html/vendor/craftcms/cms/src/console/controllers/MigrateController.php(360): yii\console\controllers\BaseMigrateController->migrateUp()
#7 [internal function]: craft\console\controllers\MigrateController->actionAll()
#8 /home/user/domain.com/public_html/vendor/yiisoft/yii2/base/InlineAction.php(57): call_user_func_array()
#9 /home/user/domain.com/public_html/vendor/yiisoft/yii2/base/Controller.php(178): yii\base\InlineAction->runWithParams()
#10 /home/user/domain.com/public_html/vendor/yiisoft/yii2/console/Controller.php(182): yii\base\Controller->runAction()
#11 /home/user/domain.com/public_html/vendor/yiisoft/yii2/base/Module.php(552): yii\console\Controller->runAction()
#12 /home/user/domain.com/public_html/vendor/yiisoft/yii2/console/Application.php(180): yii\base\Module->runAction()
#13 /home/user/domain.com/public_html/vendor/craftcms/cms/src/console/Application.php(90): yii\console\Application->runAction()
#14 /home/user/domain.com/public_html/vendor/yiisoft/yii2/console/Application.php(147): craft\console\Application->runAction()
#15 /home/user/domain.com/public_html/vendor/craftcms/cms/src/console/Application.php(121): yii\console\Application->handleRequest()
#16 /home/user/domain.com/public_html/vendor/yiisoft/yii2/base/Application.php(384): craft\console\Application->handleRequest()
#17 /home/user/domain.com/public_html/craft(22): yii\base\Application->run()
#18 {main}
*** failed to apply m220308_100000_owners_table (time: 0.114s)
```

### Steps to reproduce

1. Update from Craft Cms `3.7.51` to `^4.0` (4.2.5.2) and super-table plugin from `2.7.2` to `^3.0` (3.0.2)
2. Run migrations (`php ./craft migrate/all`)

### Craft CMS version

4.2.5.2

### Plugin version

3.0.2

### Multi-site?

Yes

**Comment by ctangney-tulip** [comment]
+1 -- I am experiencing the same exact behavior as @MarcelWeidum 

My replication steps are identical.

## Craft CMS version
4.1.2

## Plugin version
3.0.3

## Multi-site?
Yes

**Comment by ctangney-tulip** [comment]
After a little digging into the error `Integrity constraint violation: 1452 Cannot add or update a child row: a foreign key constraint fails` a little more, the best I could surmise was that the migration was attempting to populate the `$ownersTable` with a record that did not exist from the `$blocksTable`

I was able to remediate this error by adjusting lines 35-40 of the file `./src/migrations/m220308_100000_owners_table.php` from:

```php
$this->execute(<<<SQL
    INSERT INTO $ownersTable ([[blockId]], [[ownerId]], [[sortOrder]]) 
    SELECT [[id]], [[ownerId]], [[sortOrder]] 
    FROM $blocksTable
    SQL
);
```

to the following (inclusion of IGNORE for malformed / missing records):

```php
$this->execute(<<<SQL
    INSERT IGNORE INTO $ownersTable ([[blockId]], [[ownerId]], [[sortOrder]])
    SELECT [[id]], [[ownerId]], [[sortOrder]]
    FROM $blocksTable
    SQL
);
```

After some spot checking across my site, and validating a few entries via GraphQL, it seems that the migration ran successfully. If this looks like a valid fix for the above, I would be more than happy to submit a pull request for this.

@MarcelWeidum @engram-design

**Comment by engram-design** [comment]
Fixed in [3.0.4](https://github.com/verbb/super-table/releases/tag/3.0.4)
